rllib/algorithms/ppo/torch/ppo_torch_learner.py

```diff
@@ -44,7 +45,23 @@ def compute_loss_for_module(
         # TODO (Kourosh): We may or may not user module_id. For example if we have an
         # agent based learning rate scheduler, we may want to use module_id to get the
         # learning rate for that agent.
-        # TODO (Kourosh): come back to RNNs later
+
+        # RNN case: Mask away 0-padded chunks at end of time axis.
+        if self.module[module_id].is_recurrent():
+            # In the RNN case, we expect incoming tensors to be padded to the maximum
+            # sequence length. We infer the max sequence length from the actions
+            # tensor.
+            maxlen = torch.max(batch[SampleBatch.SEQ_LENS])
+            mask = sequence_mask(batch[SampleBatch.SEQ_LENS], maxlen=maxlen)
+            num_valid = torch.sum(mask)
+
+            def possibly_masked_mean(t):
+                return torch.sum(t[mask]) / num_valid
```

Comment:
This is akin to what we did in the old PPO loss.
I've enhanced the sequence mask method to support BxT batches.